openedx/core/djangoapps/schedules/content_highlights.py

```diff
@@ -154,23 +143,27 @@ def _get_highlights_for_week(sections, week_num, course_key):
     return section.highlights
 
 
-def _get_highlights_for_next_section(course_module, sections, start_date, target_date):
-    for index, section, weeks_to_complete in spaced_out_sections(course_module):
-        if not _section_has_highlights(section):
-            continue
-
+def _get_highlights_for_next_section(course, start_date, target_date):
+    """ Using the target date, retrieves highlights for the next section. """
+    use_next_sections_highlights = False
+    for index, section, weeks_to_complete in spaced_out_sections(course):
         # We calculate section due date ourselves (rather than grabbing the due attribute),
         # since not every section has a real due date (i.e. not all are graded), but we still
         # want to know when this section should have been completed by the learner.
         section_due_date = start_date + weeks_to_complete
 
-        if section_due_date.date() == target_date and index + 1 < len(sections):
-            # Return index + 2 for "week_num", since weeks start at 1 as opposed to indexes,
-            # and we want the next week, so +1 for index and +1 for next
-            return sections[index + 1].highlights, index + 2
+        if section_due_date.date() == target_date:
+            use_next_sections_highlights = True
+        elif use_next_sections_highlights and not _section_has_highlights(section):
+            raise CourseUpdateDoesNotExist(
+                'Next section [{}] has no highlights for {}'.format(section.display_name, course.id)
+            )
+        elif use_next_sections_highlights:
+            return section.highlights, index + 1
 
-    raise CourseUpdateDoesNotExist(
-        u"No section found ending on {} for {}".format(
-            target_date, course_module.id
```

Comment:
Removed this and switched to `return None, None` because now that we check _all_ schedules within a certain time period, this scenario is going to happen very often and logging it out would be unhelpful.